lib/fixed-points.nix

```diff
@@ -1,26 +1,66 @@
 { lib, ... }:
 rec {
   /*
-    Compute the fixed point of the given function `f`, which is usually an
-    attribute set that expects its final, non-recursive representation as an
-    argument:
+    `fix f` computes the fixed point of the given function `f`. In other words, the return value is `x` in `x = f x`.
 
+    `f` is usually returns an attribute set that expects its final, non-recursive representation as an argument.
+    `f` must be a lazy function.
```

Comment:
```suggestion
    `fix f` computes `x` such that `x = f x`, also known as the fixed point of `f`.
    In other words, it calls `f` with the result of itself.
    This is only meaningful when `x` is a value that can be partially evaluated, such as an attribute set, a lists or a function.
    This way, `f` can use one part of `x` to compute another part.
```

I'd suggest a wording something like this.

An example with a list and a function would also be good.

Reply:
This is great!

```suggestion
    `fix f` computes `x` such that `x == f x`, also known as the fixed point of `f`.

    This is only meaningful when `x` is a value that can be partially evaluated, such as an attribute set, a list, or a function.
    This way, `f` can use one part of `x` to compute another part.
```

I'd remove the "in other words" because it misleads the reader into tying brain knots.

Reply:
I'd rather you don't edit my own comment fwiw! (why does GitHub even allow that though..) I edited it back now.

In particular, the "This can only be meaningful when" part is intentional, it should not be "This is only meaningful when". Because just using e.g. an attribute set in the result doesn't mean you'll get something "meaningful", e.g. `f (attrs: { x = attrs.x + 1; })`. 

The only thing we know here is that if you _don't_ use an attribute set, a list or a function, it's definitely going to either result in an infinite recursion (`f (x: x + 1)`), or in a constant value (`f (x: 1)`), which is definitely not meaningful.